### ISSUE TITLE
Embed videos and formulas

### ISSUE THREAD

**Issue by heruan** [vaadin/vaadin-rich-text-editor#107]
In the QuillJS [homepage](https://quilljs.com) the editor can embed videos and formulas (there are dedicated toolbar buttons for that). I don't know Quill's internals and how it has been implemented as a Vaadin component, but would it be possible to add these missing features?

**Comment by web-padawan** [comment]
Thanks for the issue. We are currently using custom build of Quill which does not include all the features, to make bundle size smaller. It should be possible to add those features, the only reason why we decided to go with the limited scope is actually the challenges related to the Safari.

**Comment by heruan** [comment]
Thank you for the response! Not sure what do you mean by "the challenges related to the Safari", maybe support for old Safari versions?

I don't know how much does a full-featured Quill impact on the bundle size, but I think it would be desirable by the most to be able to use all the editor features and customize the toolbar.

**Comment by web-padawan** [comment]
No, that's related to the latest Safari versions supporting shadow DOM but not `ShadowRoot.getSelection()`. We are using unofficial pollyfill for it (and that's another reason for custom Quill build) but it appears to be flaky in certain Edge cases.

See the full story here: https://github.com/webcomponents/shadydom/issues/113#issuecomment-427066346
